src/shared/codelens/localLambdaRunner.ts

```diff
@@ -0,0 +1,335 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as path from 'path'
+import * as tcpPortUsed from 'tcp-port-used'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+import { buildHandlerConfig, getLocalLambdaConfiguration, HandlerConfig } from '../../lambda/local/configureLocalLambda'
+import { detectLocalLambdas } from '../../lambda/local/detectLocalLambdas'
+import { CloudFormation } from '../cloudformation/cloudformation'
+import { mkdir, writeFile } from '../filesystem'
+import { SamCliBuildInvocation } from '../sam/cli/samCliBuild'
+import { SamCliProcessInvoker, SamCliTaskInvoker } from '../sam/cli/samCliInvoker'
+import { SamCliLocalInvokeInvocation } from '../sam/cli/samCliLocalInvoke'
+import { SettingsConfiguration } from '../settingsConfiguration'
+import { SamTemplateGenerator } from '../templates/sam/samTemplateGenerator'
+import { ExtensionDisposableFiles } from '../utilities/disposableFiles'
+
+import { getLogger, OutputChannelName } from './codeLensUtils'
+
+const localize = nls.loadMessageBundle()
+
+const logger = getLogger(OutputChannelName.Lambda)
+
+export interface LambdaLocalInvokeArguments {
+  document: vscode.TextDocument,
+  range: vscode.Range,
+  handlerName: string,
+  debug: boolean,
+  workspaceFolder: vscode.WorkspaceFolder
+}
+
+export interface SAMTemplateEnvironmentVariables {
+  [resource: string]: {
+      [key: string]: string
+  }
+}
+
+export class LocalLambdaRunner {
+
+  private static readonly TEMPLATE_RESOURCE_NAME: string = 'awsToolkitSamLocalResource'
+  private static readonly SAM_LOCAL_PORT_CHECK_RETRY_INTERVAL_MILLIS: number = 125
+  private static readonly SAM_LOCAL_PORT_CHECK_RETRY_TIMEOUT_MILLIS_DEFAULT: number = 30000
+
+  private _baseBuildFolder?: string
+  private readonly _debugPort?: number
+
+  public constructor(
+      private readonly configuration: SettingsConfiguration,
+      private readonly localInvokeArgs: LambdaLocalInvokeArguments,
+      debugPort: number | undefined,
```

Comment:
Any particular reason not to declare this field inline with the rest?

Reply:
Which field do you mean? 

I attempted to minimize changes to `LocalLambdaRunner` and just migrated the existing code from [typescriptCodeLensProvider](https://github.com/aws/aws-toolkit-vscode/blob/develop/src/shared/codelens/typescriptCodeLensProvider.ts) to it's own file so it made more sense to re-use for other languages. Agreed though, there are simplifications that can be made.

Reply:
> Which field do you mean?

`debugPort`/`_debugPort`

Not a big deal, just curious.